### ISSUE TITLE
add ssm.isActive

### ISSUE THREAD

**Issue by jonathan-fielding** [SimpleStateManager/SimpleStateManager#18]
Allow checking if a state is currently active

**Comment by wlanni** [comment]
Can you confirm that this works? I get errors trying to use `isActive()`:
```
> ssm.getState('desktop');
e {id: "desktop", query: "(min-width: 768px)", options: Object, valid: true, active: true…}

> ssm.isActive('desktop');
VM13011:2 Uncaught TypeError: ssm.isActive is not a function
```

**Comment by jonathan-fielding** [comment]
Thanks for getting in touch,

This original ticket was against 2.0.0 I believe, when the library was rewritten from scratch for 3.0.0 it was not reimplemented, I can add this very easily though, can you give me some details of your use case please? would be good to understand how the library is being used by other developers

Thanks

Jonathan

**Comment by wlanni** [comment]
Oh sure, and sorry I haven't branched and added myself. It's late right now, but maybe tomorrow if I have time I can give it a shot as well. Here's the use case though:

We've add states for an extra small max (max-width: 768px) and a small min (min-width: 769px) (think Bootstrap breakpoints). We've named them id: 'xs-max' and id: 'sm-min' respectively.

We've got Bootstrap "collapse" panels on the page. They are only supposed to function in xs-max, and stop functioning in sm-min. To accomplish this, .on('click') I want to check the active state and if xs-max is active, allow for the default bootstrap collapse functionality. If not, stopPropogation();

I expected ssm.getActive('sm-min') to function, but it did not, as per your comment. :)

To work around, I used the getState() and checked the .active property, which totally worked:

```javascript
$('#product-features-content').on('click', 'h2[data-toggle="collapse"]', function(e) {
	if( ssm.getState('sm-min').active === true ) {
		e.stopPropagation();
	} else {
		$(this).find('.toggle-visibility span').toggleClass('hidden');
	}
});
```

In looking at your src/ssm.js I figured I could extend your StateManager.prototype and do something similar to your getState method, except add in checking if the active property is true. 

If you'd like, I can do this at some point this week--one of my goals this year is to contribute into some projects, and it would be a great little exercise for me, esp. if you're busy. As I wrote above, I've worked around not having it, so it's not a show-stopper.

**Comment by jonathan-fielding** [comment]
Hi @wlanni thanks for your patience,

 I have been terrible this week with getting back to people, (currently actively maintaining about 10 open source projects)

So if you download master from the repo this feature should now exist, will do more testing and add unit tests over weekend and hopefully get a proper release out soon. I did pretty much what you said, used the existing getState method and checked its active property

Thanks for help, 

Jonathan

**Comment by jonathan-fielding** [comment]
This feature has been released in 3.1.0 sorry for delay
